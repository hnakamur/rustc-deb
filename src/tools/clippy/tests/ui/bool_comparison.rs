#[warn(clippy::bool_comparison)]
fn main() {
    let x = true;
    if x == true {
        "yes"
    } else {
        "no"
    };
    if x == false {
        "yes"
    } else {
        "no"
    };
    if true == x {
        "yes"
    } else {
        "no"
    };
    if false == x {
        "yes"
    } else {
        "no"
    };
    if x != true {
        "yes"
    } else {
        "no"
    };
    if x != false {
        "yes"
    } else {
        "no"
    };
    if true != x {
        "yes"
    } else {
        "no"
    };
    if false != x {
        "yes"
    } else {
        "no"
    };
    if x < true {
        "yes"
    } else {
        "no"
    };
    if false < x {
        "yes"
    } else {
        "no"
    };
    if x > false {
        "yes"
    } else {
        "no"
    };
    if true > x {
        "yes"
    } else {
        "no"
    };
    let y = true;
    if x < y {
        "yes"
    } else {
        "no"
    };
    if x > y {
        "yes"
    } else {
        "no"
    };
}
