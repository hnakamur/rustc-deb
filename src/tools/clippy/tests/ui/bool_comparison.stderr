error: equality checks against true are unnecessary
  --> $DIR/bool_comparison.rs:4:8
   |
LL |     if x == true {
   |        ^^^^^^^^^ help: try simplifying it as shown: `x`
   |
   = note: `-D clippy::bool-comparison` implied by `-D warnings`

error: equality checks against false can be replaced by a negation
  --> $DIR/bool_comparison.rs:9:8
   |
LL |     if x == false {
   |        ^^^^^^^^^^ help: try simplifying it as shown: `!x`

error: equality checks against true are unnecessary
  --> $DIR/bool_comparison.rs:14:8
   |
LL |     if true == x {
   |        ^^^^^^^^^ help: try simplifying it as shown: `x`

error: equality checks against false can be replaced by a negation
  --> $DIR/bool_comparison.rs:19:8
   |
LL |     if false == x {
   |        ^^^^^^^^^^ help: try simplifying it as shown: `!x`

error: inequality checks against true can be replaced by a negation
  --> $DIR/bool_comparison.rs:24:8
   |
LL |     if x != true {
   |        ^^^^^^^^^ help: try simplifying it as shown: `!x`

error: inequality checks against false are unnecessary
  --> $DIR/bool_comparison.rs:29:8
   |
LL |     if x != false {
   |        ^^^^^^^^^^ help: try simplifying it as shown: `x`

error: inequality checks against true can be replaced by a negation
  --> $DIR/bool_comparison.rs:34:8
   |
LL |     if true != x {
   |        ^^^^^^^^^ help: try simplifying it as shown: `!x`

error: inequality checks against false are unnecessary
  --> $DIR/bool_comparison.rs:39:8
   |
LL |     if false != x {
   |        ^^^^^^^^^^ help: try simplifying it as shown: `x`

error: less than comparison against true can be replaced by a negation
  --> $DIR/bool_comparison.rs:44:8
   |
LL |     if x < true {
   |        ^^^^^^^^ help: try simplifying it as shown: `!x`

error: greater than checks against false are unnecessary
  --> $DIR/bool_comparison.rs:49:8
   |
LL |     if false < x {
   |        ^^^^^^^^^ help: try simplifying it as shown: `x`

error: greater than checks against false are unnecessary
  --> $DIR/bool_comparison.rs:54:8
   |
LL |     if x > false {
   |        ^^^^^^^^^ help: try simplifying it as shown: `x`

error: less than comparison against true can be replaced by a negation
  --> $DIR/bool_comparison.rs:59:8
   |
LL |     if true > x {
   |        ^^^^^^^^ help: try simplifying it as shown: `!x`

error: order comparisons between booleans can be simplified
  --> $DIR/bool_comparison.rs:65:8
   |
LL |     if x < y {
   |        ^^^^^ help: try simplifying it as shown: `!x & y`

error: order comparisons between booleans can be simplified
  --> $DIR/bool_comparison.rs:70:8
   |
LL |     if x > y {
   |        ^^^^^ help: try simplifying it as shown: `x & !y`

error: aborting due to 14 previous errors

